builtin/rebase.c

```diff
@@ -166,5 +186,45 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 		break;
 	}
 
-	return !!run_specific_rebase(&options);
+	options.upstream_name = argv[1];
+	if (!strcmp(options.upstream_name, "-"))
+		options.upstream_name = "@{-1}";
+
+	options.upstream = peel_committish(options.upstream_name);
+	if (!options.upstream)
+		die(_("invalid upstream '%s'"), options.upstream_name);
+
+	/* Make sure the branch to rebase onto is valid. */
+	options.onto_name = options.onto ?
```

Comment:
BTW this was where we looked at https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html.